### PR TITLE
Add contract reader

### PR DESCRIPTION
**Pull request by jazz-cb**
Add a simple app to demonstrate contract reads. An edit: the demo supports both sepolia and mainnet

https://github.com/user-attachments/assets/17ca7a87-0058-4d5f-8533-9fe74980ad81

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: app/api/contractreader/route.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    const result = await readContract({
+      networkId: network,
+      abi: abi,
+      contractAddress: contractAddress as `0x${string}`,
+      method: method,
+      args: args,
+    });
+
+    // If the result is a BigInt, convert it to a string
+    const resultStr = typeof result === 'bigint' ? result.toString() : result;
```
Comment (rohan-agarwal-coinbase) [inline app/api/contractreader/route.ts:18]: i think there may be a problem here if you read a function that returns a nested type, where a nested type is `bigint`.

File: package.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
   "dependencies": {
     "@aave/core-v3": "^1.19.3",
     "@aave/deploy-v3": "^1.56.2",
-    "@coinbase/coinbase-sdk": "^0.8.0",
+    "@coinbase/coinbase-sdk": "^0.9.0",
```
Comment (rohan-agarwal-coinbase) [inline package.json:14]: probably doesn't matter to be 0.9.1 for that type issue, but can we bump to keep versions up to date in any case?

Comment (rohan-agarwal-coinbase) [inline package.json:14]: probably doesn't matter to be 0.9.1 for that type issue, but can we bump to keep versions up to date in any case?

**Review by rohan-agarwal-coinbase — APPROVED** [review]

**Pull request merged**